spec/src/main/asciidoc/messaging-spec.adoc

```diff
@@ -26,3 +26,7675 @@ include::license-efsl.adoc[]
+ does thefollowing:
+
+* Use JNDI to find a ConnectionFactory object
+* Use JNDI to find one or more Destination objects
+* Use the ConnectionFactory to create a JMSContext object
+* Use the JMSContext to create the JMSProducer and JMSConsumer objects
+needed.
+* Delivery of messages is started automatically
+
+At this point a client has the basic Jakarta Messaging setup needed to 
+produce and consume messages.
+
+=== Security
+
+Jakarta Messaging does not provide features for controlling or 
+configuring message integrity or message privacy.
+
+It is expected that many Jakarta Messaging providers will provide such 
+features. It is also expected that configuration of these services will 
+be handled by provider-specific administration tools. Clients will get 
+the proper security configuration as part of the administered objects 
+they use. 
+=== Multi-threading
+
+Jakarta Messaging could have required that all its objects support 
+concurrent use. Since support for concurrent access typically adds some 
+overhead and complexity, the Jakarta Messaging design restricts its 
+requirement for concurrent access to those objects that would naturally 
+be shared by a multi-threaded client. The remaining objects are designed
+ to be accessedby one logical thread of control at a time.
+
+Table 2‑2 Objects used in the classic API, showing which support
+concurrent use
+
+[cols=",",options="header",]
+|===
+|Jakarta Messaging Object |Supports Concurrent Use
+|Destination |YES
+|ConnectionFactory |YES
+|Connection |YES
+|Session |NO
+|MessageProducer |NO
+|MessageConsumer |NO
+|===
+
+Table 2‑3 Objects used in the simplified API, showing which support
+concurrent use
+
+[cols=",",options="header",]
+|===
+|Jakarta Messaging Object |Supports Concurrent Use
+|Destination |YES
+|ConnectionFactory |YES
+|JMSContext |NO
+|JMSProducer |NO
+|JMSConsumer |NO
+|===
+
+Table 2‑4 Objects used in the domain-specific API for point-to-point
+messaging, showing which support concurrent use
+
+[cols=",",options="header",]
+|===
+|Jakarta Messaging Object |Supports Concurrent Use
+|Destination |YES
+|QueueConnectionFactory |YES
+|QueueConnection |YES
+|QueueSession |NO
+|QueueSender |NO
+|QueueReceiver |NO
+|===
+
+Table 2‑5 Objects used in the domain-specific API for pub/sub messaging,
+showing which support concurrent use
+
+[cols=",",options="header",]
+|===
+|Jakarta Messaging Object |Supports Concurrent Use
+|Destination |YES
+|TopicConnectionFactory |YES
+|TopicConnection |YES
+|TopicSession |NO
+|TopicPublisher |NO
+|TopicSubscriber |NO
+|===
+
+Jakarta Messaging defines some specific rules that restrict the 
+concurrent use of sessions. These apply to the Session object in the 
+classic API and to the QueueSession and TopicSession objects in the 
+domain-specific APIs. They also apply to the JMSContext object in the 
+simplified API since it encompasses a session. Since these rules require
+ more knowledge of Jakarta Messagingspecifics than we have presented at 
```

Comment:
There's a gratuitous spec at the beginning of the line and a missing space in the middle between two words.